# Week 8 : Case Study 4 - Bimal H M

## Testing of Hypothesis

### STEP 1 : Importing all Necessary Libraries in to Notebook

In [1]:
import pandas as pd
import numpy as np
import scipy.stats as stats

### STEP 2 : Reading the Dataset in to Notebook

In [2]:
data= pd.read_csv('Sales_add.csv')

### STEP 3 : Checking the Dataset

#### A) Displaying Dataset Info

In [3]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 22 entries, 0 to 21
Data columns (total 5 columns):
 #   Column                          Non-Null Count  Dtype 
---  ------                          --------------  ----- 
 0   Month                           22 non-null     object
 1   Region                          22 non-null     object
 2   Manager                         22 non-null     object
 3   Sales_before_digital_add(in $)  22 non-null     int64 
 4   Sales_After_digital_add(in $)   22 non-null     int64 
dtypes: int64(2), object(3)
memory usage: 1008.0+ bytes


#### B) Displaying First few Rows

In [4]:
data.head()

Month      Region      Manager  Sales_before_digital_add(in $)  \
0  Month-1  Region - A  Manager - A                          132921   
1  Month-2  Region - A  Manager - C                          149559   
2  Month-3  Region - B  Manager - A                          146278   
3  Month-4  Region - B  Manager - B                          152167   
4  Month-5  Region - C  Manager - B                          159525   

   Sales_After_digital_add(in $)  
0                         270390  
1                         223334  
2                         244243  
3                         231808  
4                         258402

#### C) Taking a descriptive analysis on dataset

In [5]:
data.describe()

Sales_before_digital_add(in $)  Sales_After_digital_add(in $)
count                       22.000000                      22.000000
mean                    149239.954545                  231123.727273
std                      14844.042921                   25556.777061
min                     130263.000000                  187305.000000
25%                     138087.750000                  214960.750000
50%                     147444.000000                  229986.500000
75%                     157627.500000                  250909.000000
max                     178939.000000                  276279.000000

### Q1 : The company wishes to clarify whether there is any increase in sales after stepping into digital marketing.

<b> To check whether there is an increase in sales or not, we can conduct a paired t_test on Sales_before_digital_add and Sales_After_digital_add <b>

#### A) Stating of Hypothesis and setting Significance Level
<pre>Let

H0 (Null Hypothesis)        : There is no significant difference in sales even after entering into digital marketing.

H1 (Alternative Hypothesis) : There is a significant increase in sales after stepping into digital marketing.

------

and Let the Level of significance be <b> &alpha; = 0.05 </b></pre>

#### B) Calculating test statistic and corresponding P-Value

In [6]:
stats.ttest_rel(data['Sales_After_digital_add(in $)'],data['Sales_before_digital_add(in $)'],alternative='greater')

Ttest_relResult(statistic=12.09070525287017, pvalue=3.168333502287889e-11)

#### C) Conclusion

since our calculated p value = 3.168 * 10^(-11) is less than our hypothesised p value = 0.05 , That is    P_calculated < P_assumed , Hence at 0.05 level of significance we can reject our Null Hypothesis (H0) and may accept H1.

Hence we can conclude that there is a significant increase in sales after stepping in to digital marketing.

### Q2 : The company needs to check whether there is any dependency between the features “Region” and “Manager”

<b> To check whether there is a dependency between the features 'Region' and ' Manager' we can conduct a Chi-square test of independence of variables with a contingency table </b>

#### A) Stating of Hypothesis and setting Significance Level
<pre>Let

H0 (Null Hypothesis)        : There is no dependency between Region and Manager

H1 (Alternative Hypothesis) : There is a relationship between Region and Manager

------

and Let the Level of significance be <b> &alpha; = 0.05 </b></pre>

#### B) Creating a contingency table of Region and Manager

In [7]:
table = pd.crosstab(data.Manager,data.Region)

In [8]:
table                    #Our Observed Values

Region       Region - A  Region - B  Region - C
Manager                                        
Manager - A           4           4           1
Manager - B           3           1           3
Manager - C           3           2           1

#### C) Calculating test statistic and corresponding P-Value

In [9]:
stats.chi2_contingency(table)

(3.050566893424036,
 0.5493991051158094,
 4,
 array([[4.09090909, 2.86363636, 2.04545455],
        [3.18181818, 2.22727273, 1.59090909],
        [2.72727273, 1.90909091, 1.36363636]]))

#### D) Conclusion

since our calculated p value = 0.549399105 is greter than our hypothesised p value = 0.05 , That is P_calculated > P_assumed , Hence at 0.05 level of significance and 4 defrees of freedom we can not reject our Null Hypothesis (H0), thus we may accept our H0 at 0.05 level of significance.
Hence we can conclude that there is no dependency between the features Region and Manager.